src/utils/image_utils.py

```diff
@@ -0,0 +1,90 @@
+"""
+Module: image_utils.py
+
+
+Contains functions that are used to analyze images.
+"""
+
+import itertools
+from typing import List, Optional
+
+from PIL import Image
+
+from src.common._io import ImageCropSelection
+
+
+def image_crop_walk(
```

Comment:
Maybe worth bringing up now, but probably not worth it holding up merging: Why do we need to walk over an image? Is it ok for two images to be overlapping? Why not randomly walk, which might prevent us from needing to hold everything in memory?

Reply:
Hey @gwaybio, good question. I am still learning on how to use `pillow` 😅 ! 

Once I have a better grasp on how `pillow` works, I may be able to answer your questions through a different PR!  For now, I will create an issue in regard to your comments and get back to it when I can implement these features!

Reply:
Issue posted!: #2

Reply:
perfect.